cmd/fabric8-jenkins-idler/main.go

```diff
@@ -45,19 +74,47 @@ func main() {
 		}
 		os.Exit(1)
 	}
+	return config
+}
 
-	// Create Toggle (Unleash) Service client
+func createFeatureToggle(config configuration.Configuration) toggles.Features {
+	var err error
 	var features toggles.Features
 	if len(config.GetFixedUuids()) > 0 {
-		log.Infof("Using fixed UUID list for toggle feature: %s", config.GetFixedUuids())
+		mainLogger.Infof("Using fixed UUID list for toggle feature: %s", config.GetFixedUuids())
 		features, err = toggles.NewFixedUUIDToggle(config.GetFixedUuids())
 	} else {
 		features, err = toggles.NewUnleashToggle(config.GetToggleURL())
 	}
 	if err != nil {
-		log.Fatal(err)
+		// Fatal with exit program
+		mainLogger.WithField("err", err).Fatal("Unable to create feature toggles")
 	}
+	return features
+}
 
-	idler := NewIdler(config, features)
-	idler.Run()
+func osioToken(config configuration.Configuration) string {
```

Comment:
So far whatever, go code that I have written I have avoided using log.Fatal
So if something can throw error, I would return the error instead. 
But which way is correct, is debatable. And I think using log.Fatal is a practice used throughout this project. If you were to change, you will have to change a lot of things. So I think we can avoid that.

Reply:
I agree. It is actually not hard to change. It is only here in main.go and this is pretty much also the only place where I would think such an approach makes sense.